### PR TITLE
Consolidate scenario

### PR DESCRIPTION
**Pull request by cournape**
@johntyree should remove any reference from `enstaller.new_solver.yaml_utils`. I also backported the tests and one feature from the code in enstaller.

### CODE REVIEW

File: simplesat/test_utils.py [hunk 1]
```diff
@@ -143,3 +147,4 @@ def print_solution(self, pool, positive_decisions):
             package = pool._id_to_package[package_id]
             print("{}: {} {}".format(package_id, package.name,
                                      package.full_version))
+
```
Comment (johntyree) [inline simplesat/test_utils.py]: Accidental extra line?

File: simplesat/tests/test_test_utils.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            "numpy 1.8.1-1; depends (MKL ~= 10.3)",
+            "numpy 1.8.1-2; depends (MKL ~= 10.3)",
+        ]
+        repository_packages = [
+            "MKL 10.3-1",
+            "numpy 1.8.1-2",
+        ]
+        repository_info = BroodRepositoryInfo("https://acme.com", "acme/loony")
+        r_numpy = P(
+            "numpy 1.8.1-2; depends (MKL ~= 10.3)", repository_info,
```
Comment (johntyree) [inline simplesat/tests/test_test_utils.py]: What does `~=` do differently than `==` in this context?

Comment (cournape) [inline simplesat/tests/test_test_utils.py]: `~=` defines a special matching comparator for which two versions are equal iff their upstream version matches. E.g. `1.9.2-1` and `1.9.2-2` match w/ `~=`, but `1.8.2-1` and `1.9.2-1` don't.

**Comment by johntyree** [comment]
LGTM

**Pull request merged**